File: proof_of_concept/test-vista.pl
#!/usr/bin/perl -w
use PseudoVistA;
# If you don't know the code, don't mess around below -BCIV
# Purpose: programatically create an appointment based on the next available clinic time in
# The Department of Veterans Affairs VistA EHR or whatever the latest buzzword for a system
# that contains patient data.
# Written by: Will BC Collins IV {a.k.a., 'BCIV' follow me on GitHub ~ I'm super cool.}
# Email: william.collins@va.gov
# Last Modified: 20151120 unless I forgot to update this line...

my $pv=new PseudoVistA;
my $connect='false';
# see ../config/sample.demo.config for example
$pv->configure('../config/dev.config');

$pv->csession(); # or $pv->connect(); for ssh connections

$pv->{exp}->expect($pv->{timeout},
  [ qr/The authenticity of host/ => sub { $pv->xsend("y\n"); }],
  [ qr/Password/i => sub { $pv->xsend("$pv->{password}\n"); }],
  [ qr/ACCESS CODE:/=>sub{ $pv->xsend("$pv->{access_code}\r"); }],
  [ qr/VERIFY CODE:/=>sub{ $pv->xsend("$pv->{verify_code}\r"); }],
  [ qr/Select TERMINAL TYPE NAME:/=>sub{ $pv->xsend("\r"); }],
  [ qr/Select Systems Manager Menu <TEST ACCOUNT> Option:/=>sub{
    $connect='true';
    #$pv->xsend("^^MAS MASTER MENU\r");
    $pv->xsend("^\r\r\r");
  }],
);

if($connect eq 'true'){
    print "\n\nSuccessfully connected to VistA!\n\n";
}
else{
    print "\n\nConnection to VistA failed.\n\n";
}

File: proof_of_concept/test-vista.rb
#!/usr/bin/ruby

require 'pty'
require 'expect'

fnames = []
PTY.spawn("csession cache -UVISTA '^ZU'") do |r_f,w_f,pid|
   w_f.sync = true

   $expect_verbose = true

   #r_f.expect(/The authenticity of host/) do
   #  w_f.print "y\n"
   #end

   r_f.expect(/^ACCESS CODE: /) do
     w_f.print "innovat3\r"
   end

   r_f.expect(/^VERIFY CODE: /) do
     w_f.print "innovat3.\r"
   end

   r_f.expect(/^Select TERMINAL TYPE NAME: /) do
     w_f.print "C-VT100\r"
   end

   r_f.expect(/Option: /) do
     #w_f.print "innovat3\n"
     print "\n-----------------------\n"
     print "Successfully logged in.\n"
     print "-----------------------\n"
     w_f.print "\r\r\r"
   end

#   if !ENV['USER'].nil?
#     username = ENV['USER']
#   elsif !ENV['LOGNAME'].nil?
#     username = ENV['LOGNAME']
#   else
#     username = 'guest'
#   end

#   r_f.expect('word:') do
#     w_f.print username+"@\n"
#   end
#   r_f.expect("> ") do
#     w_f.print "cd pub/ruby\n"
#   end
#   r_f.expect("> ") do
#     w_f.print "dir\n"
#   end

#   r_f.expect("> ") do |output|
#     for x in output[0].split("\n")
#       if x =~ /(ruby.*\.tar\.gz)/ then
#          fnames.push $1
#       end
#     end
#   end
   begin
     w_f.print "quit\n"
   rescue
   end
end
